core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -133,7 +133,8 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
    */
   private val replayExecutor: ExecutorService = {
     if (!conf.contains("spark.testing")) {
-      ThreadUtils.newDaemonSingleThreadExecutor("log-replay-executor")
+      val numProcessors = Runtime.getRuntime.availableProcessors()
```

Comment:
I would rather see this configurable.  Many times history server runs on same machine as other things (like Yarn ResourceManager or history server for MapReduce, etc) and I wouldn't want the history server to starve out more important things.